packages/react-input/src/components/Input/useInputStyles.ts

```diff
@@ -40,46 +48,110 @@ const useRootStyles = makeStyles({
   base: theme => ({
     display: 'flex',
     alignItems: 'center',
-    flexWrap: 'no-wrap',
-    gap: horizontalSpacing.xxs,
+    flexWrap: 'nowrap',
+    ...shorthands.gap(horizontalSpacing.xxs),
     fontFamily: theme.fontFamilyBase,
-    borderRadius: theme.borderRadiusMedium, // used for all but underline
+    ...shorthands.borderRadius(theme.borderRadiusMedium), // used for all but underline
+    position: 'relative',
     boxSizing: 'border-box',
-    '*, *:before, *:after': {
+
+    // This is all for the bottom focus border.
+    // It's supposed to be 2px flat all the way across and match the radius of the field's corners.
+    ':after': {
       boxSizing: 'border-box',
+      content: '""',
+      position: 'absolute',
+      left: '-1px',
+      bottom: '-1px',
+      right: '-1px',
+
+      // Maintaining the correct corner radius:
+      // Use the whole border-radius as the height and only put radii on the bottom corners.
+      // (Otherwise the radius would be automatically reduced to fit available space.)
+      // max() ensures the focus border still shows up even if someone sets theme.borderRadiusMedium to 0.
+      height: `max(2px, ${theme.borderRadiusMedium})`,
+      borderBottomLeftRadius: theme.borderRadiusMedium,
+      borderBottomRightRadius: theme.borderRadiusMedium,
+
+      // Flat 2px border:
+      // By default borderBottom will cause little "horns" on the ends. The clipPath trims them off.
+      // (This could be done without trimming using `background: linear-gradient(...)`, but using
+      // borderBottom makes it easier for people to override the color if needed.)
+      borderBottomWidth: '2px',
+      borderBottomStyle: 'solid',
+      borderBottomColor: theme.colorCompoundBrandStroke,
+      clipPath: 'inset(calc(100% - 2px) 0 0 0)',
+
+      // Animation for focus OUT
+      transform: 'scaleX(0)',
+      transition: `transform ${motionDurations.ultraFast} ${motionCurves.accelerateMid}`,
+    },
+    ':focus-within:after': {
+      // Animation for focus IN
+      transform: 'scaleX(1)',
+      transition: `transform ${motionDurations.normal} ${motionCurves.decelerateMid}`,
+    },
+    ':focus-within:active:after': {
+      // This is if the user clicks the field again while it's already focused
+      borderBottomColor: theme.colorCompoundBrandStrokePressed,
     },
   }),
   small: theme => ({
     minHeight: fieldHeights.small,
-    padding: `0 ${horizontalSpacing.sNudge}`,
+    ...shorthands.padding(`0 ${horizontalSpacing.sNudge}`),
```

Comment:
```suggestion
    ...shorthands.padding(0, horizontalSpacing.sNudge),
```

Reply:
Oops, I was trying to separate the arguments but missed that one